modules/SocketManager/module/src/socketmanager.c

```diff
@@ -380,30 +380,37 @@ ind_soc_run_status_set(ind_soc_run_status_t s)
 
 
 /*
- * Return the time in ms until the next timer could fire, or -1 if no timers
- * are active.
+ * Return the time in ms until the next timer could fire
+ * or SOCKETMANAGER_CONFIG_TIMER_PEEK_MS. A value of 0 is returned if an invalid
+ * value of time is calculated
  */
 static int
 find_next_timer_expiration(indigo_time_t now)
 {
     if (!list_empty(&ready_timers)) {
         return 0;
     }
-
     timer_wheel_entry_t *entry =
         timer_wheel_peek(timer_wheel, now + SOCKETMANAGER_CONFIG_TIMER_PEEK_MS);
     if (entry) {
-        /* if the timer is late getting serviced, return 0 */
-        if (now > entry->deadline) {
-            AIM_LOG_TRACE("find_next_timer_expiration deadline late:",
+        /*
+         * There are two out of bounds cases to handle:
+         * 1) entry->deadline < now : We fetched an expiration timer which
+         *    should have occurred before now. Hence we need to choose a value
+         *    of 0. "entry->deadline - now" has a value > INT64_MAX.
+         * 2) entry->deadline - now > INT32_MAX : In this case the delta has a
+         *    value which cannot be represented in int32_t. We should return 0
+         * returning 0 for delta > INT32_MAX handles the above two cases.
+         */
+        uint64_t delta = entry->deadline - now;
+        if (delta > INT32_MAX) {
```

Comment:
i no longer think this is correct.

The return value of '0' is to indicate that the timeout for poll should be '0' ... ie: return immediately after identifying active fd's from the fd's passed in.

'0' means that when timer's are checked, one of them is ready to deliver.

When 'delta > INT32_MAX' that mor ... or may not be true.

When 'now > entry->deadline', then we should return zero.

But when delta > INT32_MAX,  this means there is a timer value which can't be expressed by in, 
*however no timer is ready to be delivered*

If no timer is ready to be delivered, and '0' is returned, then the poll won't block, no timer will be called, and the loop will be iterated repeatedly until some other event fires.

here's what i think is correct"
```
     if (now > entry->deadline) {
        return 0;
     }
     uint64_t delta = entry->deadline - now:
     if (delta > INT32_MAX) {
        return INT32_MAX;
     }